### ISSUE TITLE
patchwork doesn't work inside electron

### ISSUE THREAD

**Issue by dominictarr** [ssbc/patchwork-electron#14]
I have been having reports of patchwork not starting up right,
and tested my self and confirmed this is a bug.

patchwork starts, but does not detect any saved history - taking you straight to the setup flow.

once you are setup, it never replicates anything, just showing an empty inbox.

I see that `patchwork-electron` is still using `electron-prebuilt@0.37` could that be the problem?

**Comment by dominictarr** [comment]
okay: the same thing happens if you install patchwork and then try to run it with electron,
and _also_ if you start it with electron and then open the UI with the browser you'll get the same thing in the browser. seems like somehow it isn't creating the connection with sbot correctly, and is behaving like it's empty instead of an error.

when you are running it with electron, `sbot whoami` also fails, but it should work because it should just connect to the sbot server that is inside of patchwork

**Comment by dominictarr** [comment]
hmm... but weird thing is that I can run `cd scuttlebot; electron bin.js server` and that works as a correct sbot server...

**Comment by dominictarr** [comment]
okay, disabling js crypto helps, at least that lets you connect to sbot

**Comment by dominictarr** [comment]
nevermind, that is working for me now...
okay. I think I've figured out what the problem is!

leveldb isn't being accessed from inside electron. therefore it's acting as if the database is empty

**Comment by dominictarr** [comment]
okay, I have figured this out!

you need correctly paired node and electron versions.
node@6.2 => electron-prebuilt@1.1
node@6.4 => electron-prebuilt@1.2
and there does not seem to be a node version that works with `electron-prebuilt@1.3`

@clehner what version is this built with? because it's not working for people

**Comment by dominictarr** [comment]
okay I think I got release to build, using `electron-prebuilt@1.1` and `node@6.2` I'm on arch so I can't test the .deb, but the tar.gz worked correctly.

**Comment by dominictarr** [comment]
okay maybe that is wrong, it's only working with electron-prebuilt@1.1 and not later

**Comment by clehner** [comment]
Last I built a release it was with node@5 and electron-prebuilt@0.36.12

It might be a good iea to update to using node 6 for everything.  Some 
things might need to be updated to handle electron API changes though.

**Comment by dominictarr** [comment]
I have successfully ran patchwork with 6, I think we can upgrade. It just looks like we need test the binary before publishing, is the release manual or is the upload part of the script?

**Comment by dominictarr** [comment]
also @mmckegg what versions where you buliding with?

**Comment by clehner** [comment]
I had a script to automate building and uploading a patchwork-electron 
release but I don't have access to that right now.

Here are the basic steps I have been using for doing a release. For 
building and publishing patchwork, I started using a local staging repo 
with dependencies all locally installed (not symlinked), to avoid 
various problems.
- in patchwork dev repo:
  - check that it works
  - tag a release
- in patchwork staging repo:
  - pull changes from local dev repo
  - run `npm i; npm publish`
  - push commits and tags to remote
- in patchwork-electron repo:
  - update versions in `app/packag.json`
  - tag the release
  - run `npm run release`
  - push commits and tags
  - upload built archives to the github release
- post release notes on ssb

**Comment by mmckegg** [comment]
@dominictarr I actually can't remember. I may have had an old version of node installed at the time. I recall running `npm install` and then `npm run release` and it just worked. 

Are these reports about the packaged version or when you use this repo directly?

**Comment by dominictarr** [comment]
@mmckegg what node version are you running currently? I had people reporting that they installed this and then nothing happened - just empty patchwork - and I found some versions combinations of electron and node just don't work. it runs, but it can't access the database correctly.

**Comment by mmckegg** [comment]
@dominictarr running 6.3.1 right now.

So they installed the DMG version and there was just an empty database? Odd, I'll try upgrading my node and see what happens.

**Comment by dominictarr** [comment]
@mmckegg I tried it on my already running database, and I just failed to read anything in the database and acted like it was an empty database!

**Comment by dominictarr** [comment]
can you build the latest one, and just check that it actually runs

**Comment by mmckegg** [comment]
Just trying that now!

**Comment by mmckegg** [comment]
So it initially failed with the `Version mismatch` error, but I rebuilt it using [require-rebuild](https://github.com/juliangruber/require-rebuild) and then everything seemed to work as expected. 

I also tried it with ssb-patchwork v2.10.1 and it worked with the feed loading properly. In addition, I built the DMG and tried to run it from there, and that worked perfectly too!

## 

All I did was add `require-rebuild` as a dev dependency to `app/package.json` and then require it from `app/index.js`:

``` js
require('require-rebuild')()
require('ssb-patchwork')
```

**Comment by dominictarr** [comment]
awesome, can we upload the releases? I want to check if it works

**Comment by mmckegg** [comment]
@dominictarr I just created a draft release for v2.10.1 and uploaded my DMG build. Hopefully you'll be able to see the files.

https://github.com/ssbc/patchwork-electron/releases/tag/untagged-2ed6f56d22f0bb9f4e00

**Comment by dominictarr** [comment]
thanks! I'll try that out and see how it goes!

**Comment by mmckegg** [comment]
When I was trying to prepare the latest Mac build, I did some experimentation with different electron versions.

I discovered that version v1.2 seems to have the problem mentioned above, however v1.1 is fine (hence why the latest build uses v1.1.3 of electron). I also did [this](https://github.com/ssbc/patchwork-electron/pull/15) to make it easy to test different electron versions without reinstalling node.
